File: libs/minimagick/reprocess_simple.rb
require "mini_magick"

picturePath = ARGV[0]
driver = ARGV[1]
testName = ARGV[2];
width = ARGV[3];
height = ARGV[4];

if driver != "magick"
	MiniMagick.configure do |config|
		config.cli = :graphicsmagick # or :imagemagick or :imagemagick7
		driver = "graphicsmagick"
	end
end

image = MiniMagick::Image.open(picturePath)
image.path
image.resize height+"x"+width
if File.extname(picturePath) === ".png"
	image.format "png"
	image.write "./output/"+testName+"-minimagick-"+driver+".png"
else
	image.format "jpg"
	image.write "./output/"+testName+"-minimagick-"+driver+".jpg"
end

File: libs/sharp/reprocess_simple.js
const sharp = require('sharp');

var path = process.argv[2];
var driver = process.argv[3];
var testName = process.argv[4];
var width = process.argv[5];
var height = process.argv[6];
var img_extension = path.match(/\.(\w+)/)[1];
sharp(path)
  .resize(width/1, height/1)
  .toFile('./output/'+testName+'-sharp-'+driver+'.'+img_extension, (err, info) => { 
  	console.log(err)
  });